Northstar.Custom/mod/scripts/vscripts/sh_message_utils.gnut

```diff
@@ -0,0 +1,510 @@
+{
+	switch ( args[0].tointeger() )
+	{
+		case eMessageType.POLL:
+			thread ShowPollMessage_Threaded()
+			break
+		case eMessageType.LARGE:
+			client.largeMessageQueue.append( client.temp )
+			break
+		case eMessageType.POPUP:
+			client.popupMessageQueue.append( client.temp )
+			break
+		case eMessageType.ANNOUNCEMENT:
+			client.announcementQueue.append( client.temp )
+			break
+		case eMessageType.INFO:
+			client.infoMessageQueue.append( client.temp )
+			break
+		case eMessageType.CREATE_STATUS:
+			CreateStatusMessage( client.id )
+			break
+		case eMessageType.EDIT_STATUS:
+			EditStatusMessage( client.id )
+			break
+		case eMessageType.DELETE_STATUS:
+			thread DeleteStatusMessage( client.id )
+			break
+	}
+}
+
+void function DeleteStatusMessage( string id )
+{
+	if ( id in client.statusMessageList )
+	{
+		var rui = client.statusMessageList[ id ]
+		RuiSetGameTime( rui, "startFadeOutTime", Time() )
+		
+		// Remove it from table
+		delete client.statusMessageList[ id ]
+		
+		// Wait for animation
+		wait 0.6
+		
+		RuiDestroyIfAlive( rui )
+		
+		int i = 0
+		foreach( _id, _rui in client.statusMessageList )
+		{
+			RuiSetInt( _rui, "listPos", i )
+
+			i++
+		}
+	}
+}
+
+void function EditStatusMessage( string id )
+{
+	if( id in client.statusMessageList )
+	{
+		var rui = client.statusMessageList[ id ]
+		RuiSetString( rui, "titleText", client.temp.title )
+		RuiSetString( rui, "itemText", client.temp.description )
+	}
+}
+
+void function CreateStatusMessage( string id )
+{
+	// Cap at 4 messages at a time
+	if( client.statusMessageList.len() == STATUS_MESSAGES_MAX )
+		return
+
+	var rui = CreatePermanentCockpitRui( $"ui/at_wave_intro.rpak" )
+	RuiSetInt( rui, "listPos", client.statusMessageList.len() )
+	RuiSetGameTime( rui, "startFadeInTime", Time() )
+	RuiSetString( rui, "titleText", client.temp.title )
+	RuiSetString( rui, "itemText", client.temp.description )
+	RuiSetFloat2( rui, "offset", <0,-250,0> )
+
+	client.statusMessageList[ id ] <- rui
+}
+
+void function SelectPollOption_Threaded( int index )
+{
+	if ( index >= client.poll.ruis.len() || index <= 0 )
+		return
+	
+	RuiSetFloat3( client.poll.ruis[ index ], "msgColor", ColorSelected )
+	EmitSoundOnEntity( GetLocalClientPlayer(), "menu_accept" )
+	
+	float endTime = 1 + client.poll.duration
+	while( endTime > Time() && client.poll.pollActive )
+		WaitFrame()
+	
+	GetLocalClientPlayer().ClientCommand( "poll_respond " + index )
+	
+	foreach( var rui in client.poll.ruis )
+		RuiDestroyIfAlive( rui )
+	
+	client.poll.ruis.clear()
+	client.poll.pollActive = false
+}
+
+void function ShowPollMessage_Threaded()
+{
+	if( client.poll.pollActive )
+		return
+	
+	client.poll.pollActive = true
+	
+	
+	for( int i = 0; i < client.poll.options.len() + 1; i++ )
+	{
+		var rui = CreateCockpitRui( $"ui/cockpit_console_text_top_left.rpak" )
+		// This makes it fade and me no likey >:(
+		//RuiSetInt( rui, "maxLines", 1 )
+		//RuiSetInt( rui, "lineNum", i )
```

Comment:
```suggestion
```
dont need those comments do we? if you want those as a guide on how shit works maybe make it comprehensive n stuff